### PR TITLE
accounts/abi, accounts/scwallet: avoid duplicate regex compilation

### PR DESCRIPTION
**Pull request by yincongcyincong**
every time function **NewType** and **wallet.Open** execute, these regexp are initialized. which situation reduce the performance.

### CODE REVIEW

File: accounts/scwallet/wallet.go [hunk 1]
```diff
@@ -73,6 +73,14 @@ var (
 	DerivationSignatureHash = sha256.Sum256(common.Hash{}.Bytes())
 )
 
+var (
+	// PinRegexp is the regular expression used to validate PIN codes.
+	PinRegexp = regexp.MustCompile(`^[0-9]{6,}$`)
+
+	// PunRegexp is the regular expression used to validate PUN codes.
+	PunRegexp = regexp.MustCompile(`^[0-9]{12,}$`)
```
Comment (rjl493456442) [inline accounts/scwallet/wallet.go:81]: Looks like it should be PukRegexp?

Comment (yincongcyincong) [reply accounts/scwallet/wallet.go]: i changed this variable name, thanks

**Review by MariusVanDerWijden — APPROVED** [review]

**Review by rjl493456442 — APPROVED** [review]

File: accounts/scwallet/wallet.go [hunk 2]
```diff
@@ -73,6 +73,14 @@ var (
 	DerivationSignatureHash = sha256.Sum256(common.Hash{}.Bytes())
 )
 
+var (
+	// PinRegexp is the regular expression used to validate PIN codes.
+	PinRegexp = regexp.MustCompile(`^[0-9]{6,}$`)
+
+	// PukRegexp is the regular expression used to validate PUK codes.
+	PukRegexp = regexp.MustCompile(`^[0-9]{12,}$`)
```
Comment (karalabe) [inline accounts/scwallet/wallet.go:81]: Please make these private types (i.e. lowercase)

Comment (yincongcyincong) [reply accounts/scwallet/wallet.go]: i changed variable

**Review by karalabe — APPROVED** [review]
LGTM

**Pull request merged**